extensions/interactions/NumberWithUnits/directives/number-with-units-rules.service.ts

```diff
@@ -46,15 +46,27 @@ export class NumberWithUnitsRulesService {
     // Returns true only if input is exactly equal to answer.
     var answerObject = this.unitsObjectFactory.fromDict(answer);
     var inputsObject = this.unitsObjectFactory.fromDict(inputs.f);
+    let valueIsEqual: boolean;
 
-    var answerString = answerObject.toMathjsCompatibleString();
-    var inputsString = inputsObject.toMathjsCompatibleString();
+    if (answerObject.type !== inputsObject.type) {
```

Comment:
I don't see tests for all these cases -- different types, real comparison, etc.?

Reply:
Added test for it, Done.